crates/loro-internal/src/handler.rs

```diff
@@ -1130,16 +1131,22 @@ impl Handler {
                             index: _,
                             position,
                         } => {
-                            let new_target = x.__internal__next_tree_id();
                             if let Some(p) = parent.as_mut() {
                                 remap_tree_id(p, container_remap)
                             }
-                            if x.create_at_with_target_for_apply_diff(parent, position, new_target)?
-                            {
-                                container_remap.insert(
-                                    target.associated_meta_container(),
-                                    new_target.associated_meta_container(),
-                                );
+                            remap_tree_id(&mut target, container_remap);
+                            if x.contains(target) {
```

Comment:
It's better to include this case and the related design here in case we forget the background.

Should we test whether the target is alive here?